scipy/stats/_qmc.py

```diff
@@ -914,13 +910,37 @@ class LatinHypercube(QMCEngine):
 
         .. versionadded:: 1.8.0
 
+    orthogonal_array_p : int, optional
+        Orthogonal array based LHS of strength 2 [7]_, [8]_.
+        `orthogonal_array_p` must be a prime. It also constrains
+        ``d <= orthogonal_array_p + 1`` and ``n = orthogonal_array_p**2``.
+
+        .. versionadded:: 1.8.0
+
```

Comment:
It sounds like `orthogonal_array_p` must be prime and constrains the shape of the array. When it is provided, the arrays produced by `LatinHypercube` will be orthogonal with strength 2.

But what _is_ `orthogonal_array_p`?

Reply:
> It sounds like `orthogonal_array_p` must be prime and constrains the shape of the array. When it is provided, the arrays produced by `LatinHypercube` will be orthogonal with strength 2.
> 
> But what _is_ `orthogonal_array_p`?

It's the "order of the galois field". I can try to explain this but might be too involved no? Although just saying we need a prime without saying why could raise questions.

> What is going to happen if the user creates the `LatinHypercube` with `orthogonal_array_p=5` and then calls `random(100)`? If it's going to raise an error, why not just have some sort of boolean `orthogonal` option. Then whenever the user calls `random`, `orthogonal_array_p` is the square root of the argument. If it is not prime, then the error is raised.

True I could move the logic down. Then I would also need to move the check on the dimension there. In the end it might be better for the user as it would fail once. Because the current implementation can fail at the engine instantiation and also during sampling. I will change it. (Doing so, I might not need to explain what is a galois field).

> Either way, will we allow the `optimization` option for orthogonal Latin hypercubes? (Haven't thought carefully, but I'm guessing it would no longer be orthogonal, even using the `random-cd` strategy for optimization?) If that's the case, should the `optimization` option be re-named so that the user can _either_ specify that the Latin hypercube is to be optimized _or_ they want an orthogonal Latin hypercube?

Good question. For now I thought that we could allow optimization (centered is working as well, there is a test for that too). In both cases, the result is a LHS. I thought that it would be better to have all parameters doing something when used together instead of having some parameter which are not used in some cases. So one could want to optimize an orthogonal array which would even used centered coordinates.

Reply:
That's fine, too. It just needs to be clear in the documentation that the result is no longer guaranteed to be orthogonal.

Reply:
I updated the logic, let me know what you think.